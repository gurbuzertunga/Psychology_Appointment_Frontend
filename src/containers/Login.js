import { connect } from 'react-redux';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import LoginForm from '../components/loginForm';
import loginRequest from '../requests/loginRequest';
import {
  createToken,
  createUser,
  closeModal,
  openModal,
  isLoadingStarted,
} from '../actions/index';

const Login = props => {
  const {
    createUser,
    createToken,
    closeModal,
    openModal,
    isLoadingStarted,
  } = props;
  const history = useHistory();
  const handleSubmit = async data => {
    isLoadingStarted();
    const { name, email } = data;
    createUser({ name, email });
    const doctor = await loginRequest(createToken, data);
    switch (doctor) {
      case 'Invalid credentials':
        history.push('/wrong');
        break;
      case true:
        history.push('/admin');
        break;
      default:
        history.push('/consultancies');
        break;
    }
    closeModal();
    isLoadingStarted();
  };
  return (
    <div className="flex flex-col">
      <h1 className="text-center my-4 text-white text-2xl font-extrabold">Login</h1>
      <LoginForm handleClick={handleSubmit} />
      <div className="font-light text-center lg:text-base text-2xl cursor-pointer" onClick={() => openModal('signup')} onKeyDown={openModal} role="presentation">Create a new Account</div>
    </div>
  );
};
Login.propTypes = {
  createToken: PropTypes.func,
  isLoadingStarted: PropTypes.func,
  createUser: PropTypes.func,
  closeModal: PropTypes.func,
  openModal: PropTypes.func,
  history: PropTypes.shape({
    push: PropTypes.func,
  }),
};
Login.defaultProps = {
  createToken,
  createUser,
  closeModal,
  openModal,
  isLoadingStarted,
  history: {},
};

export default connect(null, {
  createToken,
  createUser,
  closeModal,
  openModal,
  isLoadingStarted,
})(Login);
