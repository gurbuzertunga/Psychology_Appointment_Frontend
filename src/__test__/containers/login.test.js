import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Login from '../../containers/Login';

configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <Login />
  </Provider>,
);

describe('Login Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the Consultancies List Component', () => {
    const page = component.find('.consultancyList');
    expect(page).toMatchSnapshot();
  });
});
