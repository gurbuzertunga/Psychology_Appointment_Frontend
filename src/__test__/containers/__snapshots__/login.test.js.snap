// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders the Consultancies List Component 1`] = `ShallowWrapper {}`;
